// react-simple-tab/js/simple-tab.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _simpleTabLabels = require('./simple-tab-labels');

var _simpleTabLabels2 = _interopRequireDefault(_simpleTabLabels);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var SimpleTab = function (_Component) {
  _inherits(SimpleTab, _Component);

  function SimpleTab() {
    _classCallCheck(this, SimpleTab);

    var _this = _possibleConstructorReturn(this, Object.getPrototypeOf(SimpleTab).call(this));

    _this.state = {
      currentTab: 0
    };
    return _this;
  }

  _createClass(SimpleTab, [{
    key: 'onTabChange',
    value: function onTabChange(index) {
      this.setState({ currentTab: index });
    }
  }, {
    key: 'render',
    value: function render() {
      var tabContent = this.props.tabs.length > 0 ? this.props.tabs[this.state.currentTab].content : null;

      return React.createElement(
        'div',
        { style: this.props.style },
        React.createElement(_simpleTabLabels2.default, {
          tabs: this.props.tabs,
          style: this.props.labelListStyle,
          onTabChange: this.onTabChange.bind(this),
          currentTab: this.state.currentTab }),
        tabContent
      );
    }
  }]);

  return SimpleTab;
}(_react.Component);

SimpleTab.propTypes = {
  tabs: _react.PropTypes.array.isRequired,
  style: _react.PropTypes.object,
  labelListStyle: _react.PropTypes.object
};
exports.default = SimpleTab;


module.exports = SimpleTab;
